src/wp-includes/taxonomy.php

```diff
@@ -1943,6 +1943,11 @@ function wp_delete_term( $term, $taxonomy, $args = array() ) {
 	$object_ids = (array) $wpdb->get_col( $wpdb->prepare( "SELECT object_id FROM $wpdb->term_relationships WHERE term_taxonomy_id = %d", $tt_id ) );
 
 	foreach ( $object_ids as $object_id ) {
+		if ( ! isset( $default ) ) {
+			wp_remove_object_terms( $object_id, $term, $taxonomy );
+			continue;
+		}
+
```

Comment:
This is a great idea.

A unit test or two to make sure it's never broken would be dandy. If `$default` gets changed in some way so it's always set, it would be a shame to go back.

Maybe I am drunk, but could `wp_defer_term_counting()` help if there is a default? If one exists and is added to multiple objects then the term count takes place on a per object basis.